ex2_system/source/leop/leop.c

```diff
@@ -107,20 +118,16 @@ bool deploy_all_deployables() {
  * @return void
  */
 bool execute_leop() {
-    // TODO: When eeprom is working, use commended code
     bool eeprom_flag = false;
     eeprom_flag = eeprom_get_leop_status();
     if (eeprom_flag != true) {
-        // If leop sequence was never executed, check that all hard switches have been deployed
-        if (deploy_all_deployables() == true) {
-            // If all hard switch have been deployed, set eeprom flag to TRUE
-            eeprom_set_leop_status();
-            return true;
-        }
-        return false;
-    }
-    //    else if (eeprom_get_leop_status() == true) {
-    else if (eeprom_flag == true) {
+        // If leop sequence was never executed, doit
+        deploy_all_deployables();
```

Comment:
Why did you remove all the error handling? If LEOP doesn't work, we don't want to indicate that it did

Reply:
The final switch statuses are all we really care about, no? There should be no reason for leop to not complete. Whether or not it was successful, we will know by the switch statuses

Reply:
I think the error handling can stay, we can remove the return false statements though so LEOP can proceed onto the next switch. We should keep the logs in error handling to understand what went wrong and when.

Reply:
What's the purpose of error handling if it'll always return true?

I don't understand what utility it provides us to have logs in between deployment attempts. We just care about the final state after LEOP is done. What would we do with the other information from in between attempts? How would that inform our actions post-leop?

Reply:
The same reason for including so much information in our beacon. Generally speaking it's better to have more information. It's also a good practice to know how many attempts were made and how many failures we saw in the process, this is a standard typically followed in the industry, and maybe Robert can speak to this more but I think it should be a standard we follow in our code. I prefer to have a complete sequence of events, even if the final outcome is the same.